include/psa/crypto.h

```diff
@@ -3759,6 +3759,418 @@ psa_status_t psa_generate_key(const psa_key_attributes_t *attributes,
+ *                             output.
+ *
+ * \retval #PSA_SUCCESS
+ *         Success.
+ * \retval #PSA_ERROR_BAD_STATE
+ *         The operation state is not valid (it must be active, but beyond that
+ *         validity is specific to the protocol).
+ * \retval #PSA_ERROR_BUFFER_TOO_SMALL
+ *         The size of the \p output buffer is too small.
+ * \retval #PSA_ERROR_INSUFFICIENT_MEMORY
+ * \retval #PSA_ERROR_COMMUNICATION_FAILURE
+ * \retval #PSA_ERROR_HARDWARE_FAILURE
+ * \retval #PSA_ERROR_CORRUPTION_DETECTED
+ * \retval #PSA_ERROR_STORAGE_FAILURE
+ * \retval #PSA_ERROR_BAD_STATE
+ *         The library has not been previously initialized by psa_crypto_init().
+ *         It is implementation-dependent whether a failure to initialize
+ *         results in this error code.
+ */
+psa_status_t psa_pake_output(psa_pake_operation_t *operation,
+                             psa_pake_data_t type,
+                             uint8_t *output,
+                             size_t output_size,
+                             size_t *output_length);
+
+/** Provide peer key share to a password-authenticated key exchange operation.
+ *
+ * This function inputs a simple key share (eg. group element).
+ *
+ * The exact sequence of calls to perform a password-authenticated key
+ * exchange depends on the protocol in use.  Refer to the documentation of
+ * individual PAKE algorithm types (`PSA_ALG_XXX` values of type
+ * ::psa_algorithm_t such that #PSA_ALG_IS_PAKE(\c alg) is true) for more
+ * information.
+ *
+ * \param[in,out] operation       Active PAKE operation.
+ * \param[in] key_share           Buffer containing the peer's key share.
+ * \param key_share_length        Size of the \p key_share buffer in bytes.
+ *
+ * \retval #PSA_SUCCESS
+ *         Success.
+ * \retval #PSA_ERROR_BAD_STATE
+ *         The operation state is not valid (it must be active, but beyond that
+ *         validity is specific to the protocol).
+ * \retval #PSA_ERROR_INSUFFICIENT_MEMORY
+ * \retval #PSA_ERROR_COMMUNICATION_FAILURE
+ * \retval #PSA_ERROR_HARDWARE_FAILURE
+ * \retval #PSA_ERROR_CORRUPTION_DETECTED
+ * \retval #PSA_ERROR_STORAGE_FAILURE
+ * \retval #PSA_ERROR_BAD_STATE
+ *         The library has not been previously initialized by psa_crypto_init().
+ *         It is implementation-dependent whether a failure to initialize
+ *         results in this error code.
+ */
+psa_status_t psa_pake_set_key_share(psa_pake_operation_t *operation,
+                                    const uint8_t *key_share,
+                                    size_t key_share_length);
+
+/** Provide additional peer key share for a password-authenticated key exchange.
+ *
+ * Depending on the protocol being executed, you might need to call this
+ * function several times or you might not need to call this at all.
+ *
+ * Calling this function with PSA_PAKE_DATA_KEY_SHARE as \p type is equivalent
+ * to calling psa_pake_set_key_share().
```

Comment:
I was slightly conflicted by that too and was thinking about removing it. I kept it because:
1. That was the direction we agreed on the latest meeting before I raised this PR
2. It might not make much sense for J-PAKE, but feels appropriate to more mainstream PAKEs

In summary, I am happy to remove it for now, we still can add it back later if we think it is worth it.